src/parser/smt2/smt2.h

```diff
@@ -257,6 +261,12 @@ class Smt2 : public Parser
    */
   const LogicInfo& getLogic() const { return d_logic; }
 
+  /**
+   * Add g to the list of grammars to keep track of.
+   * @param g grammar to keep track of
+   */
+  api::Grammar* addGrammar(api::Grammar g);
```

Comment:
I still prefer the name `mkGrammar`, since it indicates that the parser owns it.
Another option would be `allocateGramar`.
The name "addGrammar" feels like the user of this method is the one that owns it, which is misleading.

Reply:
Done!